Purchases/Purchasing/ProductsManager.swift

```diff
@@ -31,7 +31,7 @@ class ProductsManager: NSObject {
 
         if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *),
            SystemInfo.useStoreKit2IfAvailable {
```

Comment:
ooo do we want to wrap (some of)  showManageSub and beginRefundRequest with this? or is this intended more for purchasing flows?

Reply:
I think anywhere where we're doing `Task {`, we should replace it with `_ = Task<Void, Never>`, just for future-proofing - in case we ever add a throwing method in the contents of the Task, so we can't forget to handle the error

Reply:
I did replace the existing usage for the showManageSub and beginRefundRequest logic

Reply:
> I did replace the existing usage for the showManageSub and beginRefundRequest logic

Sorry, I wasn't clear, I meant the `useStoreKit2IfAvailable1`. Unrelated to this PR..

Reply:
OOhhhhhh I see, sorry. 
For `useStoreKit2IfAvailable`, I was mostly trying to wrap the feature flag the logic to fetch products and purchase products, but both the refund request and manage subscription modals are sk2-only and there's no reason not to use them. Like, since they're new things, if SK2 has a bug, it won't break your existing flows. 
But if SK2 has a bug in purchasing, it might actually break the whole thing. 
Or even if it's fully stable, using SK2 might not be a good idea if you also want to use features that we currently only support for SK1, like discounted purchases

Reply:
> OOhhhhhh I see, sorry. For `useStoreKit2IfAvailable`, I was mostly trying to wrap the feature flag the logic to fetch products and purchase products, but both the refund request and manage subscription modals are sk2-only and there's no reason not to use them. Like, since they're new things, if SK2 has a bug, it won't break your existing flows. But if SK2 has a bug in purchasing, it might actually break the whole thing. Or even if it's fully stable, using SK2 might not be a good idea if you also want to use features that we currently only support for SK1, like discounted purchases

OK, totally fair. We might want to have a comment on the boolean (if that's the way we go) clarifying this, just in CASE people thing something like beginRefundRequest won't work without that flagged on. manage sub does have a fallback for non-sk2 though, right? so potentially this flag could mean don't use sk2.  i think your thought makes more sense and it should only be for purchasing/products, we should just be fully clear when we release

Reply:
what do you think about renaming this variable to `useStoreKit2ForPurchasesIfAvailable`?

Reply:
or maybe `useStoreKit2ForProductssIfAvailable`

Reply:
> or maybe `useStoreKit2ForProductsIfAvailable`

sure! maybe we can update that for the first beta and see if there is any feedback?